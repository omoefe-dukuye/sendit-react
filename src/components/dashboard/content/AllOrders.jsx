import React, { Component } from 'react';
import { toast } from 'react-toastify';
import AllParcels from './allParcels';
import axios from '../../../utils/axiosConfig';
import Loader from '../../loader';

export default class AllOrders extends Component {
  state = {
    loading: true
  };

  async componentDidMount() {
    try {
      const { data: { orders: parcels } } = await axios.get(`${process.env.API_ROOT_URL}/parcels`);
      this.setState({ parcels });
    } catch ({ response: { data: { error } } }) {
      toast.error(error);
    }
    this.setState({ loading: false });
  }

  render() {
    const { parcels, loading } = this.state;

    return (
      <div>
        {
          loading
            ? <Loader />
            : <AllParcels parcels={parcels} />
        }
      </div>
    )
  }
}
