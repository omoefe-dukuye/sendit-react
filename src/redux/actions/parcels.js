import { toast } from "react-toastify";
import axios from "../../utils/axiosConfig";
import { ADD_ORDER } from "./types";

const api = process.env.API_ROOT_URL;

export const createOrder = (payload) => async (dispatch) => {
  try {
    const { data: { parcel } } = await axios.post(`${api}/parcels`, payload);

    dispatch({
      type: ADD_ORDER,
      payload: parcel
    });

    return parcel;
  } catch ({ response: { data: { error } } }) {
   toast.error(error);
  }
}

export default createOrder;
