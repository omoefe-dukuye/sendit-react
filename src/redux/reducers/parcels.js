import { ADD_ORDER } from '../actions/types';

const parcelReducerDefaultState = [];

export default (state = parcelReducerDefaultState, action) => {
  switch(action.type) {
    case ADD_ORDER:
      return [...state, action.payload];

    default:
      return state;
  }
};
