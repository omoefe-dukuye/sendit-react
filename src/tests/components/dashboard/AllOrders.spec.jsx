import React from 'react';
import { shallow } from 'enzyme';
import '@babel/polyfill';
import axios from '../../../utils/axiosConfig';
import parcels from '../../mocks/mockParcels';
import AllOrders from '../../../components/adminDashboard/content/AllOrders';

jest.mock('../../../utils/axiosConfig');

test('Should render static component', async () => {
  axios.get.mockResolvedValue({ data: { orders: parcels } });
  const wrapper = await shallow(<AllOrders />);
  expect(wrapper).toMatchSnapshot();
});

test('Should handle server errors', async () => {
  axios.get.mockRejectedValue({ response: { data: { error: 'error' } } });
  const wrapper = await shallow(<AllOrders />);
  expect(wrapper).toMatchSnapshot();
});
