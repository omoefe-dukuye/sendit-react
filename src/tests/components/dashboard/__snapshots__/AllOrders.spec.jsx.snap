// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should handle server errors 1`] = `
<div>
  <_default />
</div>
`;

exports[`Should render static component 1`] = `
<div>
  <_default />
</div>
`;
